tests/checks/test_pkgdir.py

```diff
@@ -279,13 +279,79 @@ def test_multiple_files_over_limit(self):
                 f.seek(size)
                 f.write('\0')
         r = self.assertReports(self.mk_check(), [pkg])
-        assert len(r) == 2
+        assert len(r) == 3
         assert isinstance(r[0], pkgdir.SizeViolation)
         assert isinstance(r[1], pkgdir.SizeViolation)
+        assert isinstance(r[2], pkgdir.TotalSizeViolation)
         assert (
-            tuple(sorted((x.filename, x.size) for x in r)) ==
+            tuple(sorted((x.filename, x.size) for x in r[:2])) ==
             (('files/massive', 1024*100+1), ('files/over', 1024*20+1))
         )
+        assert r[2].size == 1024*(10+20+20+100)+4-1
+
+
+class TestManifestSizeViolation(PkgDirCheckBase):
+    """Check ManifestSizeViolation results."""
+
+    def write_manifest(self, pkg, size):
+        with open(pjoin(os.path.dirname(pkg.path), 'Manifest'), 'w') as f:
+            f.seek(size - 1)
+            f.write('\0')
+
+    def test_one_ebuild_no_manifest(self):
+        pkg = self.mk_pkg()
+        touch(pjoin(os.path.dirname(pkg.path), f'{pkg.package}-9999.ebuild'))
+        self.assertNoReport(self.mk_check(), [pkg])
+
+    def test_one_ebuild_below_limit(self):
+        pkg = self.mk_pkg()
+        touch(pjoin(os.path.dirname(pkg.path), f'{pkg.package}-0.ebuild'))
+        self.write_manifest(pkg, 512)
```

Comment:
Maybe it would be better to use relative size here (I know 512 is really small number, but just in case we decide to change units or something like this)? I'm not sure - so your call 😄 

```suggestion
        self.write_manifest(pkg, pkgdir.ManifestSizeViolation.limit_per_version / 2)
```